### PR TITLE
fix(cli): restore source file when reverting a move across folders

### LINKED ISSUE #9741: [Critical Bug] Revert Fails to Restore Files After AI Moves Them Between Folders — Files Deleted Not Restored
### Description

**### [Critical Bug] Revert Fails to Restore Files After AI Moves Them Between Folders — Files Deleted Not Restored**

- Revert causes irreversible data/file loss, which is the opposite of what a revert operation should guarantee

When the AI moves files from an existing folder (e.g. Folder A) to a newly created folder (e.g. Folder B), triggering Revert does not restore the original state correctly.
Instead of moving the files back to Folder A, the revert operation only deletes them from Folder B — resulting in permanent file loss with no recovery.

**Expected Behavior**

- Revert should fully restore the initial state
- Files that were moved from Folder A → Folder B should be moved back to Folder A
- Folder B (if it didn't exist before) should be removed
- No files should be lost or deleted during the revert process

**Actual Behavior**

- Files are deleted from Folder B ✅
- Files are NOT restored to Folder A ❌
- Files are permanently lost with no recovery path ❌
- The workspace is left in a broken intermediate state ❌

### Plugins

_No response_

### Kilo version

7.2.xx

### Steps to reproduce

1. Open Kilo Code with a project containing files in Folder A
2. Send a prompt that causes the AI to:
- Move files out of Folder A
- Create a new Folder B
- Place the moved files into Folder B

3. Trigger the Revert function to restore the initial state
4.Observe that files are deleted from Folder B but never copied back to Folder A

### Screenshot and/or share link

AI Action:
Folder A [file1, file2]  →  Folder B [file1, file2] (newly created)

Expected Revert:
Folder B [deleted] → Folder A [file1, file2] ✅

Actual Revert:
Folder B [deleted] → Folder A [empty] ❌  (files permanently lost)

### Operating System

Windows 25h2

### Terminal

_No response_

### PR DESCRIPTION
**Pull request by lambertjosh**
## Summary

Fixes #9741. When the agent moved a file between folders during a task, clicking Revert would delete the file from its new location without restoring the original — resulting in data loss.

**Root cause:** `patch()` in `snapshot/index.ts` computed the file list via `git diff --cached --name-only` *without* `--no-renames`. Git's default rename detection collapsed a move (`A/foo` → `B/foo`) into a single destination entry, so the source path never entered the revert list.

**Fix:** Pass `--no-renames`, matching `diffFull()`'s behavior. Both source and destination now appear, so revert deletes the new location and restores the original.

## Changes

- One-line fix in `patch()` to add `--no-renames`
- Regression tests in `test/kilocode/snapshot-revert-move.test.ts` covering folder-to-folder moves and in-folder renames
- Docs update in `checkpoints.md` and `whats-new.md` (the limitation note now documents expected behavior; kept in case there are edge cases we haven't surfaced)
- Changeset

### CODE REVIEW

**Review by marius-kilocode — APPROVED** [review]

File: packages/kilo-docs/source-links.md [hunk 1]
```diff
@@ -62,6 +62,8 @@
   <!-- packages/kilo-vscode/webview-ui/src/components/settings/AboutKiloCodeTab.tsx -->
 - <https://github.com/Kilo-Org/kilocode/issues/6986>
   <!-- packages/kilo-vscode/src/agent-manager/constants.ts -->
+- <https://github.com/Kilo-Org/kilocode/issues/9741>
```
Comment (marius-kilocode) [inline packages/kilo-docs/source-links.md]: What is this doing?

**Review by marius-kilocode — APPROVED** [review]

File: packages/kilo-docs/source-links.md (hunk 1 above)
Comment (lambertjosh) [reply packages/kilo-docs/source-links.md]: Pretty sure it's an automatic thing that grabs any links in the source files and puts them here. I don't know why. Since this issue is referenced in the comments, it gets put here. 

`<!-- Auto-generated by script/extract-source-links.ts — DO NOT EDIT -->`

Comment (lambertjosh) [reply packages/kilo-docs/source-links.md]: I've removed this.

**Comment by marius-kilocode** [comment]
@lambertjosh do you want to go ahead with this? Did you test it and is it ready to merge?

**Comment by lambertjosh** [comment]
image
image
image
image
